x-pack/plugins/security_solution/public/attack_discovery/use_attack_discovery/index.tsx

```diff
@@ -87,241 +67,99 @@ export const useAttackDiscovery = ({
-  const connectorIntervals = useMemo(
-    () => generationIntervals?.[connectorId ?? ''] ?? [],
-    [connectorId, generationIntervals]
-  );
-
-  // generation can take a long time, so we calculate an approximate future time:
-  const [approximateFutureTime, setApproximateFutureTime] = useState<Date | null>(null);
-
-  // get cached attack discoveries if they exist:
-  const [attackDiscoveries, setAttackDiscoveries] = useState<AttackDiscovery[]>(
-    cachedAttackDiscoveries[connectorId ?? '']?.attackDiscoveries ?? []
-  );
-
-  // get replacements from the cached attack discoveries if they exist:
-  const [replacements, setReplacements] = useState<Replacements>(
-    cachedAttackDiscoveries[connectorId ?? '']?.replacements ?? {}
-  );
-
-  // get last updated from the cached attack discoveries if it exists:
-  const [lastUpdated, setLastUpdated] = useState<Date | null>(
-    cachedAttackDiscoveries[connectorId ?? '']?.updated ?? null
-  );
+  const [attackDiscoveries, setAttackDiscoveries] = useState<AttackDiscoveries>([]);
+  const [replacements, setReplacements] = useState<Replacements>({});
+  const [lastUpdated, setLastUpdated] = useState<Date | null>(null);
 
   // number of alerts sent as context to the LLM:
   const [alertsContextCount, setAlertsContextCount] = useState<number | null>(null);
 
-  /** The callback when users click the Generate button */
-  const fetchAttackDiscoveries = useCallback(async () => {
+  const requestBody = useMemo(() => {
     const selectedConnector = aiConnectors?.find((connector) => connector.id === connectorId);
-    const actionTypeId = getFallbackActionTypeId(selectedConnector?.actionTypeId);
-
-    const body = getRequestBody({
-      actionTypeId,
+    const genAiConfig = getGenAiConfig(selectedConnector);
+    return getRequestBody({
       alertsIndexPattern,
       anonymizationFields,
-      connectorId,
+      genAiConfig,
       knowledgeBase,
+      selectedConnector,
       traceOptions,
     });
+  }, [
+    aiConnectors,
+    alertsIndexPattern,
+    anonymizationFields,
+    connectorId,
+    knowledgeBase,
+    traceOptions,
+  ]);
+
+  useEffect(() => {
+    if (connectorId != null && connectorId !== '') {
+      pollApi();
+      setAlertsContextCount(null);
+      setLastUpdated(null);
+      setReplacements({});
+      setAttackDiscoveries([]);
+      setGenerationIntervals([]);
+    }
+  }, [pollApi, connectorId]);
 
+  useEffect(() => {
+    setIsLoading(pollStatus === 'running');
+  }, [pollStatus]);
+
+  useEffect(() => {
+    if (pollData !== null && pollData.connectorId === connectorId) {
+      if (pollData.alertsContextCount) setAlertsContextCount(null);
+      if (pollData.updatedAt) setLastUpdated(new Date(pollData.updatedAt));
+      if (pollData.replacements) setReplacements(pollData.replacements);
+      setAttackDiscoveries(pollData.attackDiscoveries);
+      setGenerationIntervals(pollData.generationIntervals);
+    }
+  }, [connectorId, pollData]);
```

Comment:
Consider refactoring the three `useEffect`s below: 

```typescript
  useEffect(() => {
    if (connectorId != null && connectorId !== '') {
      pollApi();
      setAlertsContextCount(null);
      setLastUpdated(null);
      setReplacements({});
      setAttackDiscoveries([]);
      setGenerationIntervals([]);
    }
  }, [pollApi, connectorId]);

  useEffect(() => {
    setIsLoading(pollStatus === 'running');
  }, [pollStatus]);

  useEffect(() => {
    if (pollData !== null && pollData.connectorId === connectorId) {
      if (pollData.alertsContextCount) setAlertsContextCount(null);
      if (pollData.updatedAt) setLastUpdated(new Date(pollData.updatedAt));
      if (pollData.replacements) setReplacements(pollData.replacements);
      setAttackDiscoveries(pollData.attackDiscoveries);
      setGenerationIntervals(pollData.generationIntervals);
    }
  }, [connectorId, pollData]);
```

because it's not clear why the first and third `useEffect`s define the same `dependencies`, but in reverse order, i.e. the first one has `[pollApi, connectorId]` dependencies:

```typescript
  useEffect(() => {
    if (connectorId != null && connectorId !== '') {
      // ...
    }
  }, [pollApi, connectorId]);
```

and the second one has `[connectorId, pollData]` dependencies:

```typescript
  useEffect(() => {
    if (pollData !== null && pollData.connectorId === connectorId) {
      // ...
    }
  }, [connectorId, pollData]);
```